erts/emulator/beam/erl_arith.c

```diff
@@ -461,6 +461,70 @@ erts_mixed_plus(Process* p, Eterm arg1, Eterm arg2)
     }
 }
 
+Eterm
+erts_unary_minus(Process* p, Eterm arg)
+{
+    DECLARE_TMP(tmp_big,0,p);
+    Eterm hdr, res, zero;
+    FloatDef f;
+    dsize_t sz1, sz2, sz;
+    int need_heap;
+    Eterm* hp;
+    Sint ires;
+
+    ERTS_FP_CHECK_INIT(p);
+    switch (arg & _TAG_PRIMARY_MASK) {
+    case TAG_PRIMARY_IMMED1:
+        switch ((arg & _TAG_IMMED1_MASK) >> _TAG_PRIMARY_SIZE) {
+        case (_TAG_IMMED1_SMALL >> _TAG_PRIMARY_SIZE):
+            ires = -signed_val(arg);
```

Comment:
I believe this is correct, as before it would be equivalent to `ires = signed_val(make_small(0)) - signed_val(arg)`.